### ISSUE TITLE
No test assemblies found on the test machine matching the source filter criteria or no tests discovered matching test filter criteria. Verify that test assemblies are present on the machine and test filter criteria is correct

### ISSUE THREAD

**Issue by aliyahess** [microsoft/azure-pipelines-tasks#16943]
I'm trying to run automated tests from test plan with Azure devops. I've configured two tasks. But I have the following error when I run a test. It can't find the test assemblies.

![image](https://user-images.githubusercontent.com/114168197/191777897-31fe302f-40be-4652-a0ff-46b6481bc46e.png)

Logs : 

ExecutionAndPublish.Run : Microsoft.VisualStudio.TestService.VstestAdapter.TestsNotFoundException: No test assemblies found on the test machine matching the source filter criteria or no tests discovered matching test filter criteria. Verify that test assemblies are present on the machine and test filter criteria is correct.
The slice of type 'Execution' is 'Aborted' because of the error : Microsoft.VisualStudio.TestService.VstestAdapter.TestsNotFoundException: No test assemblies found on the test machine matching the source filter criteria or no tests discovered matching test filter criteria. Verify that test assemblies are present on the machine and test filter criteria is correct.
Test run is aborted. Logging details of the run logs.
System.Exception: The test run was aborted, failing the task.

![image](https://user-images.githubusercontent.com/114168197/191778144-0b83efe0-8734-4ecd-8e55-78f84d237eae.png)

**Comment by michaelsaxbury** [comment]
i am getting same error, did you get any further resolving it?

**Comment by cek-cek** [comment]
Getting this issue on the newly released VsTest version **17.4.0 x64**. Pipeline runs up to that point that were using **17.3.2 x64** and succeeded without any problems.

So our current fix is to lock VsTest at specific version using task:
```
- task: VisualStudioTestPlatformInstaller@1
  displayName: 'Visual Studio Test Platform Installer'
  inputs:
    versionSelector: specificVersion
    testPlatformVersion: 17.3.2
```

**Comment by Sildarud** [comment]
I'm facing with the same issue. Thanks for sharing the workaround.
